herddb-core/src/main/java/herddb/cdc/ChangeDataCapture.java

```diff
@@ -0,0 +1,320 @@
+                    transaction.tablesDefinitions.put(entry.tableName, null);
+                } else {
+                    tablesDefinitions.remove(entry.tableName, table);
+                }
+                fire(new Mutation(table, MutationType.DROP_TABLE, null, lsn, entry.timestamp), entry.transactionId);
+            }
+            break;
+            case LogEntryType.CREATE_TABLE: {
+                Table table = Table.deserialize(entry.value.to_array());
+                if (entry.transactionId > 0) {
+                    TransactionHolder transaction = transactions.get(entry.transactionId);
+                    transaction.tablesDefinitions.put(entry.tableName, table);
+                } else {
+                    tablesDefinitions.put(entry.tableName, table);
+                }
+                fire(new Mutation(table, MutationType.CREATE_TABLE, null, lsn, entry.timestamp), entry.transactionId);
+            }
+            break;
+            case LogEntryType.ALTER_TABLE: {
+                Table table = Table.deserialize(entry.value.to_array());
+                if (entry.transactionId > 0) {
+                    TransactionHolder transaction = transactions.get(entry.transactionId);
+                    transaction.tablesDefinitions.put(entry.tableName, table);
+                } else {
+                    tablesDefinitions.put(entry.tableName, table);
+                }
+                fire(new Mutation(table, MutationType.ALTER_TABLE, null, lsn, entry.timestamp), entry.transactionId);
+            }
+            break;
+            case LogEntryType.INSERT: {
+                Table table = lookupTable(entry);
+                DataAccessorForFullRecord record = new DataAccessorForFullRecord(table, new Record(entry.key, entry.value));
+                fire(new Mutation(table, MutationType.INSERT, record, lsn, entry.timestamp), entry.transactionId);
+            }
+            break;
+            case LogEntryType.DELETE: {
+                Table table = lookupTable(entry);
+                DataAccessorForFullRecord record = new DataAccessorForFullRecord(table, new Record(entry.key, entry.value));
+                fire(new Mutation(table, MutationType.DELETE, record, lsn, entry.timestamp), entry.transactionId);
+            }
+            break;
+            case LogEntryType.UPDATE: {
+                Table table = lookupTable(entry);
+                DataAccessorForFullRecord record = new DataAccessorForFullRecord(table, new Record(entry.key, entry.value));
+                fire(new Mutation(table, MutationType.UPDATE, record, lsn, entry.timestamp), entry.transactionId);
+            }
+            break;
+            case LogEntryType.BEGINTRANSACTION: {
+                transactions.put(entry.transactionId, new TransactionHolder());
+            }
+            break;
+            case LogEntryType.COMMITTRANSACTION: {
+                TransactionHolder transaction = transactions.get(entry.transactionId);
```

Comment:
is it better to remove committed TransactionHolder object from the Map ? 
I feel this Map will grow forever

Reply:
good catch!
fixed